doc/rst/technotes/libraries.rst

```diff
@@ -551,6 +554,140 @@ include the chapel runtime and standard modules for use in a non-Chapel program
 and when the library is linked to a Chapel program this leads to multiple
 definitions of these functions.
 
+Using Your Library in Multilocale Settings
+==========================================
+
+Prerequisites
+-------------
+
+Chapel supports ``--library`` when ``CHPL_COMM = gasnet``.  We intend to support
+other settings in the future, see :ref:`Other Settings` in the :ref:`Multilocale
+Caveats` section for more information.
+
+To compile a multilocale library, `ZeroMQ <http://zeromq.org/>`_ must be
+installed.
+
+If ZeroMQ is not installed in a way that enables your C compiler to find it
+easily, the environment variable ``CHPL_ZMQ_HOME`` can be set.  This environment
+variable should be set to a directory containing both an ``include`` directory
+which contains ``zmq.h`` and a ``lib`` directory which contains ``libzmq.*``.
+For example, for a directory structure:
+
+.. code-block:: text
+
+   |-- .local/
+   |    |-- include/
+   |    |    |-- zmq.h
+   |    |-- lib/
+   |    |    |-- libzmq.a
+   |    |    |-- libzmq.so
+
+``CHPL_ZMQ_HOME`` would be set to ``/absolute/path/to/.local/``.
+
+Initializing Your Multilocale Library
+-------------------------------------
+
+From the user's perspective, a strategy similar to that of single locale
+libraries is used when initializing a multilocale library.  However, like
+transitioning between a single locale executable and a multilocale one, users
+must now specify the number of locales required for the multilocale library.
+
+In C
+~~~~
+
+Users must still call ``chpl_library_init()`` before utilizing the exported
+Chapel functions.  However, the ``char* argv[]`` must now include two additional
+entries: the numlocales flag and its intended value.
+
+This can be accomplished either by explicitly adding the arguments in the C
+client program itself, or by passing them as arguments to the executable.
+
+The following example programs use the ``foo`` library.  This first example
```

Comment:
I'm not sure I would call the "original single locale client can be used" part an example, so it seems strange that it talks about "The following example programs". It could say "This example demonstrates explicitly adding the arguments in the program using the ``foo library``"